src/hotspot/share/logging/logAsyncWriter.cpp

```diff
@@ -212,19 +251,20 @@ AsyncLogWriter* AsyncLogWriter::instance() {
 void AsyncLogWriter::flush() {
   if (_instance != nullptr) {
     {
-      AsyncLogLocker locker;
+      OuterLocker locker;
```

Comment:
It is not obvious to me why we need the producer lock here. ??

Reply:
The `flush` method is also a producer, it modifies the buffer, so it needs both locks.